AMDirT/validate/domain/__init__.py

```diff
@@ -71,16 +72,29 @@ def read_schema(self, schema: Schema) -> dict:
         with open(schema, "r") as s:
             return json.load(s)
 
-    def read_dataset(self, dataset: Dataset) -> pd.DataFrame:
+    def read_dataset(self, dataset: Dataset, schema: Schema) -> pd.DataFrame:
         """ "Read dataset from file or string
         Args:
             dataset (str): Path to dataset in tsv format
+            schema (Schema): Parsed schema as dictionary (from read_schema)
         Returns:
             pd.DataFrame: Dataset as pandas dataframe
         """
+        string_to_dtype_conversions = {
+            'string': str,
+            'integer': pd.Int64Dtype(),
+        }
+        column_dtypes = {}
+        for column_keys in schema['items']['properties']:
+            parsed_schema_dtypes=schema['items']['properties'][column_keys]['type']
+            if len(parsed_schema_dtypes) == 1:
+                column_dtypes[column_keys] = schema['items']['properties'][column_keys]['type']
+            else:
+                if parsed_schema_dtypes == ['integer','null']: column_dtypes[column_keys] = string_to_dtype_conversions['integer']
+                if parsed_schema_dtypes == ['string','null']: column_dtypes[column_keys] = string_to_dtype_conversions['string']
```

Comment:
```suggestion
    coltype = schema['items']['properties'][column_keys]['type'][0]
    if coltype in string_to_dtype_conversions:
        column_dtypes[column_keys] = string_to_dtype_conversions[coltype]
    elif coltype == 'null':
        self.add_error(
            DFError(
                error = "Schema Error"
                source = column_dtypes[column_keys]
                column = column_keys
                row = None
                message = "Default/first type of column in schema can not be null"

        )
    else:
        column_dtypes[column_keys] = coltype
```

As null should not be the default column type in the schema, we should only care for the first element of the type list